src/finite_elasticity_routines.f90

```diff
@@ -4008,6 +4031,14 @@ SUBROUTINE FINITE_ELASTICITY_GAUSS_CAUCHY_TENSOR(EQUATIONS_SET,DEPENDENT_INTERPO
           lambda_a=UP-(UP-LOW)/2.0_DP
 
           F_a_inv=0.0_DP
+          IF(lambda_a<TOL) THEN
+           LOCAL_ERROR="Warning: lambda_a is close to zero"
+           CALL FlagError(LOCAL_ERROR,ERR,ERROR,*999)
+!            WRITE(*,*) "UP: ", UP
```

Comment:
As I said, we first tried the eigenvalue routine, but it didn't work. However, at that time, we had some other bugs in the code, which probably caused the problems we encountered, and the problem was not the eigenvalue routine. I'll ask Thomas Klotz to change this back to the eigenvalue routine, but I don't have the time now to test this properly. Would be great if you could accept this for now.